cliquet/tests/test_permission.py

```diff
@@ -82,12 +82,27 @@ def test_backend_error_is_raised_anywhere(self):
             self.assertRaises(exceptions.BackendError, *call)
 
     def test_ping_returns_false_if_unavailable(self):
+        ping = heartbeat(self.permission)
         for patch in self.client_error_patcher:
             patch.start()
-        self.assertFalse(self.permission.ping(self.request))
+        self.assertFalse(ping(self.request))
 
     def test_ping_returns_true_if_available(self):
-        self.assertTrue(self.permission.ping(self.request))
+        ping = heartbeat(self.permission)
+        self.assertTrue(ping(self.request))
+
+    def test_ping_returns_false_if_unavailable_in_readonly_mode(self):
+        self.request.registry.settings['readonly'] = 'true'
+        ping = heartbeat(self.permission)
+        with mock.patch.object(self.permission, 'user_principals',
+                               side_effect=exceptions.BackendError(
+                                   "Segmentation fault.")):
```

Comment:
nit: This indentation is not very python-friendly

Look what happens if we leave that :D https://twitter.com/tarek_ziade/status/659466245322985472